podcasts/MiniPlayerViewController+Positioning.swift

```diff
@@ -37,11 +37,44 @@ extension MiniPlayerViewController {
         })
     }
 
+    func showMiniPlayerWithDelay() {
+        // only show if something is playing
+        if PlaybackManager.shared.currentEpisode() == nil { return }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            UIView.animate(withDuration: 0.2, animations: { () in
+                self.moveToShownPosition()
+            }, completion: { _ in
+                self.moveToShownPosition() // call this again in case the animation block wasn't called. It's ok to call this twice
+                NotificationCenter.postOnMainThread(notification: Constants.Notifications.miniPlayerDidAppear)
+            })
+        }
+    }
+
     func openFullScreenPlayer(completion: (() -> Void)? = nil) {
         guard PlaybackManager.shared.currentEpisode() != nil else { return }
 
         if playerOpenState == .open || playerOpenState == .animating { return }
 
+        guard !FeatureFlag.newPlayerTransition.enabled else {
+            playerOpenState = .animating
+            aboutToDisplayFullScreenPlayer()
+
+            fullScreenPlayer?.modalPresentationStyle = .overCurrentContext
+
+            presentFromRootController(fullScreenPlayer!, animated: true) {
```

Comment:
Can we add a check instead of force unwrapping here?

There's a weird state I can put the app into that will cause a crash:
1. Add the up next widget
2. Go to Profile > Settings Cog > Private > Read privacy policy
3. Close app
4. Tap on Up next widget
5. 💥 crash

```suggestion
            presentFromRootController(fullScreenPlayer!, animated: true) {
```

Reply:
Done in cd5a9fd6731cffb3708c1667f2d6136b81c72d71

I know the nested `guard` is ugly as hell but once we remove the feature flag just one will be there so I think we're good.